packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx

```diff
@@ -1,142 +1,170 @@
+  const updateAfterMount = () => {
+    if (ref && ref.current) {
+      const scrollbar = ref.current as any;
       if (scrollbar.update) {
         scrollbar.update();
       }
     }
-  }
-
-  componentDidUpdate() {
-    this.updateScroll();
-  }
-
-  renderTrack = (track: 'track-vertical' | 'track-horizontal', hideTrack: boolean | undefined, passedProps: any) => {
-    if (passedProps.style && hideTrack) {
-      passedProps.style.display = 'none';
-    }
-
-    return <div {...passedProps} className={track} />;
   };
 
-  renderThumb = (thumb: 'thumb-horizontal' | 'thumb-vertical', passedProps: any) => {
-    return <div {...passedProps} className={thumb} />;
-  };
+  /** onMount */
+  useEffect(() => {
+    updateScroll();
 
-  renderTrackHorizontal = (passedProps: any) => {
-    return this.renderTrack('track-horizontal', this.props.hideHorizontalTrack, passedProps);
-  };
-
-  renderTrackVertical = (passedProps: any) => {
-    return this.renderTrack('track-vertical', this.props.hideVerticalTrack, passedProps);
-  };
-
-  renderThumbHorizontal = (passedProps: any) => {
-    return this.renderThumb('thumb-horizontal', passedProps);
-  };
-
-  renderThumbVertical = (passedProps: any) => {
-    return this.renderThumb('thumb-vertical', passedProps);
-  };
-
-  renderView = (passedProps: any) => {
-    return <div {...passedProps} className="view" />;
-  };
-
-  render() {
-    const {
-      className,
-      children,
-      autoHeightMax,
-      autoHeightMin,
-      setScrollTop,
-      autoHide,
-      autoHideTimeout,
-      hideTracksWhenNotNeeded,
-    } = this.props;
-
-    return (
-      <Scrollbars
-        ref={this.ref}
-        className={classNames('custom-scrollbar', className)}
-        onScroll={setScrollTop}
-        autoHeight={true}
-        autoHide={autoHide}
-        autoHideTimeout={autoHideTimeout}
-        hideTracksWhenNotNeeded={hideTracksWhenNotNeeded}
-        // These autoHeightMin & autoHeightMax options affect firefox and chrome differently.
-        // Before these where set to inherit but that caused problems with cut of legends in firefox
-        autoHeightMax={autoHeightMax}
-        autoHeightMin={autoHeightMin}
-        renderTrackHorizontal={this.renderTrackHorizontal}
-        renderTrackVertical={this.renderTrackVertical}
-        renderThumbHorizontal={this.renderThumbHorizontal}
-        renderThumbVertical={this.renderThumbVertical}
-        renderView={this.renderView}
-      >
-        {children}
-      </Scrollbars>
-    );
+    // this logic is to make scrollbar visible when content is added body after mount
+    if (ref.current === null && updateAfterMountMs) {
+      console.log('scrollbar mount updateAfterMountMs');
+      setTimeout(updateAfterMount, updateAfterMountMs);
+    }
+  });
+
+  const renderTrackHorizontal = useCallback((passedProps: any) => {
+    return renderTrack('track-horizontal', hideHorizontalTrack, passedProps);
+  }, []);
```

Comment:
This callback is dependent on `hideHorizontalTrack` so it should be part of dependencies.
```suggestion
  }, [hideHorizontalTrack]);
```

Reply:
ah, good catch!